stripe_test.go

```diff
@@ -311,17 +311,6 @@ func TestShouldRetry(t *testing.T) {
 		assert.True(t, shouldRetry)
 	})
 
-	// 500 Internal Server Error -- don't retry POST
-	t.Run("DontRetryOn500Post", func(t *testing.T) {
-		shouldRetry, _ := c.shouldRetry(
-			nil,
-			&http.Request{Method: http.MethodPost},
-			&http.Response{StatusCode: http.StatusInternalServerError},
-			0,
-		)
-		assert.False(t, shouldRetry)
```

Comment:
We could keep this test but make it `assert.True`? Or maybe change it to another 500 level like 502?

Reply:
There is a test case for 503 right after this - so I believe we are covered.

Reply:
Sounsd good!